### ISSUE TITLE
When rotating displays dimensions are not getting updated

### ISSUE THREAD

**Issue by riskable** [caemor/epd-waveshare#104]
If you call `display.set_rotation(DisplayRotation::Rotate90);` or `display.set_rotation(DisplayRotation::Rotate270);` the display's dimensions (WIDTH/HEIGHT) should be swapped so that `embedded-graphics` text alignment features can work properly.  As things are right now, if you rotate the display 90° or 270° only left-aligned text will display properly.

**Comment by riskable** [comment]
Example of how another `embedded-graphics` crate handles dimensions after rotation:

https://github.com/jamwaffles/ssd1306/blob/master/src/lib.rs#L279

...and:

https://github.com/jamwaffles/ssd1306/blob/master/src/mode/buffered_graphics.rs#L237

**Comment by riskable** [comment]
Turns out it was pretty easy to fix this... Just a bit tedious (had to make small changes to `graphics.rs` for every supported display).  Here's the confirmation that it works (this display has: `display.set_rotation(DisplayRotation::Rotate270);` applied)

**Before:**

![PXL_20220404_163738183](https://user-images.githubusercontent.com/66987/161612527-bf322ea6-e1ed-4c50-a2c7-1440f605f08a.jpg)

**After:**

![PXL_20220404_184400585](https://user-images.githubusercontent.com/66987/161612878-1b10ab5a-f586-4f23-88ac-58f41ffdc879.jpg)

**Comment by eflukx** [comment]
Yep just wanted to create this exact same issue.. and found this one.. :) 

Fixed it for my display as follows: 

```rust
impl OriginDimensions for Display2in13 {
    fn size(&self) -> Size {
        match self.rotation() {
            DisplayRotation::Rotate0 | DisplayRotation::Rotate180 => Size::new(WIDTH, HEIGHT),
            DisplayRotation::Rotate90 | DisplayRotation::Rotate270 => Size::new(HEIGHT, WIDTH),
        }
    }
}
```

If needed/wanted I could make a PR for this change

**Comment by jquesada2016** [comment]
CAme here for the same thing!

**Comment by caemor** [comment]
Should already be fixed on 0.6.
